js_modules/dagit/packages/ui/src/components/Alert.tsx

```diff
@@ -62,13 +64,20 @@ export const Alert: React.FC<Props> = (props) => {
       $textColor={textColor}
       padding={{horizontal: 16, vertical: 12}}
     >
-      <Group direction="row" spacing={12} alignItems="flex-start">
-        <IconWIP name={icon as IconName} color={iconColor} />
-        <Group direction="column" spacing={8}>
-          <AlertTitle>{title}</AlertTitle>
-          {description ? <AlertDescription>{description}</AlertDescription> : null}
+      <Box flex={{direction: 'row', justifyContent: 'space-between'}}>
+        <Group direction="row" spacing={12} alignItems="flex-start">
+          <IconWIP name={icon as IconName} color={iconColor} />
+          <Group direction="column" spacing={8}>
+            <AlertTitle>{title}</AlertTitle>
+            {description ? <AlertDescription>{description}</AlertDescription> : null}
+          </Group>
         </Group>
-      </Group>
+        {hasCloseButton ? (
```

Comment:
I think it might be slightly preferable to merge these two new props together and display the close button whenever a callback is provided (eg: `(!!onClose)` here) since either one without the other is sort of undesirable behavior.